### ISSUE TITLE
Lag at larger file sizes

### ISSUE THREAD

**Issue by SpeedyDoggo** [olivierkes/manuskript#1086]
I'm not really expecting solutions beyond 'chop up your project so it's smaller', so I figured I'd just mention the issue as a sort of stress test 
I have a single .msk file that's 171 KB and lags when I switch between text files and scroll within the text files, on both my PCs. My less powerful machine also has a delay in text rendering, but no missed inputs. My usual PC either doesn't lag or lags so slightly that it's not noticeable when I type. It doesn't do any better in focus mode or if I switch between basic and fiction mode, but the scrolling does improve if I've done it once in the text file already.

**Comment by TheJackiMonster** [comment]
This is a known issue. The problem is that grammar and spell checking will iterate over the whole text during each change/edit. There are concepts to improve this. But it will require huge changes to the implementation of the software.

For other reasons I would recommend you to not use the single-file mode. You can deactivate it in the settings. The reason for this is that saving and loading files is not 100% reliable. So it might cause corruptions. However if the project is not saved as a single compressed file, you won't lose as much in such a case. Otherwise making backups after a writing session is recommended.

**Comment by jonnners** [comment]
Aha - it's the spellchecker, thanks for that clarity. I only found it an turned it on recently, but didn't make the connection between that and why performance fell through the floor - even after I broke it up into small files it was pretty much unusable.

May I request a simple feature - put a note in the menu to advise users of the performance limitation. Presumably I can turn it on at some point, do all the checks, and turn in off again (or I might export to something else for spellchecking).

**Comment by ELECTROHERMIT** [comment]
The lag issue is a problem, is this being addressed? I do not have such an issue with large word count in any other text editing program, such as LibreOffice. In Manuskript, I turned off WORD COUNT and this helped. I do not see a means of turning off the dictionary under settings, can someone advise. IMHO my document is not so large that the program should be lagging as much as it is. I have a beefy Window10 3D workstation with 4.0Ghz Xeons, RTX cards and gobs of RAM.

**Comment by TheJackiMonster** [comment]
> The lag issue is a problem, is this being addressed? I do not have such an issue with large word count in any other text editing program, such as LibreOffice. In Manuskript, I turned off WORD COUNT and this helped. I do not see a means of turning off the dictionary under settings, can someone advise. IMHO my document is not so large that the program should be lagging as much as it is. I have a beefy Window10 3D workstation with 4.0Ghz Xeons, RTX cards and gobs of RAM.

Explaination: The applications checks your whole text file every single time you make changes to it fully against dictionary and calculates a word count dynamically as well. Because Manuskript runs in Python that can be too much. Your graphics card will also not help in that case of course. The latency from a single cpu thread is the main issue.

However we can improve this by delaying calculations asynchronously as well as improve calculations to only run on your changed portions instead of the whole file which is mostly unchanged anyway.

In other words: The issue is known and we're working on solutions for that.

**Comment by ELECTROHERMIT** [comment]
> > The lag issue is a problem, is this being addressed? I do not have such an issue with large word count in any other text editing program, such as LibreOffice. In Manuskript, I turned off WORD COUNT and this helped. I do not see a means of turning off the dictionary under settings, can someone advise. IMHO my document is not so large that the program should be lagging as much as it is. I have a beefy Window10 3D workstation with 4.0Ghz Xeons, RTX cards and gobs of RAM.
> 
> Explaination: The applications checks your whole text file every single time you make changes to it fully against dictionary and calculates a word count dynamically as well. Because Manuskript runs in Python that can be too much. Your graphics card will also not help in that case of course. The latency from a single cpu thread is the main issue.
> 
> However we can improve this by delaying calculations asynchronously as well as improve calculations to only run on your changed portions instead of the whole file which is mostly unchanged anyway.
> 
> In other words: The issue is known and we're working on solutions for that.

That is most excellent! Thank you for the update, sounds very exciting.
